plugins/inputs/interrupts/interrupts.go

```diff
@@ -112,16 +169,19 @@ func (s *Interrupts) Gather(acc telegraf.Accumulator) error {
 	for measurement, file := range map[string]string{"interrupts": "/proc/interrupts", "soft_interrupts": "/proc/softirqs"} {
 		f, err := os.Open(file)
 		if err != nil {
-			acc.AddError(fmt.Errorf("Could not open file: %s", file))
+			acc.AddError(fmt.Errorf("could not open file: %s", file))
 			continue
 		}
-		defer f.Close()
-		irqs, err := parseInterrupts(f)
+		irqs, err := parseInterrupts(f, s.Spurious)
+		_ = f.Close()
```

Comment:
The previous defer method is actually preferred, since it will be less likely to be broken in future code updates.

Reply:
The problem with the previous method is that the static code checkers complain about the `f.close()` return code being ignored, which it is. Two choices:

1. add proper func for `defer f.Close()` to explicitly ignore the return value
2. add comment for static checkers to ignore the ignored return value error

I'm fine either way. Do you have a preference?

Reply:
I'm not our goal to keep static code checkers quiet, outside of the ones required by the integration tests (go vet in particular).  So I'd rather just leave it as before, I don't find assigning to `_` to be helpful in explaining the code.

Reply:
I think I'll prefer to f.Close() in the loop. If defer is in a loop, we trigger possible leak detection because the defer's closure is the function, not the loop. In this particular location inside `Gather`, we loop twice, not a real problem. In `parseInterrupts` we open one file per irq, this can easily grow into the hundreds of open file descriptors, so it is better to close in the loop.

Reply:
Good point, this is possibly an indication that we should extract a function.

Reply:
yeah, I'm not particularly happy with the overall design, lemme try a more detailed refactor, including some improved testing